### ISSUE TITLE
A single checkpoint should not supersede a succeedingThenComplete

### ISSUE THREAD

**Issue by doctorpangloss** [vert-x3/vertx-junit5#90]
#### Describe the feature

Given code like this:

```java
@Test
void testProc(VertxTestContext testContext) {
  var checkpoint = testContext.checkpoint();
  returnsFuture()
    .onSuccess(v -> checkpoint.flag())
    .compose(v -> anotherReturnsFuture())
    .onComplete(testContext.succeedingThenComplete());
}
```

the user's intent is almost always to flag the checkpoint but to keep running until `testContext.succeedingThenComplete()` is executed.

Currently, creating a checkpoint like in the above statement will actually end the test as soon as the test thread notices `checkpoint.flag()` (i.e. the injected `testContext.awaitCompletion` will be satisfied). This is very clunky.

#### Use cases

A common use case is to run some side-by-side check independent the test code and use checkpoints to flag that your assertions were actually run. For example you might mock a class and `checkpoint.flag()` inside a mocked method that runs assertions on its arguments.

#### Contribution

I can implement this feature.

**Comment by jponge** [comment]
When there are checkpoints `TestContext` completes when they have all completed, so `onComplete` shall also flag a checkpoint.

**Comment by jponge** [comment]
Also how would you spot that the test context `succeedingThenComplete` method could eventually be called?

`succeedingThenComplete` is really a shortcut for completing immediately when one asynchronous operation completes.

**Comment by doctorpangloss** [comment]
Yes, but what happens is the test completes **before** `succeedingThenComplete` 's returned handler is called, because of something with `checkpoint.flag()`.
